File: Data Structures & Algorithms/counting-bits/submission-2.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int>output(n +1 , 0);
        for(int i = 0;i<n+1;i++){
            output[i] = output[i >> 1] + (i & 1);
        }
        return output;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        length = len(nums)
        output = [1] * length

        prefix = 1
        for i in range(length):
            output[i] = prefix
            prefix *= nums[i]
        psfix = 1
        for i in range(length -1, -1, -1):
            output[i] *= psfix
            psfix *= nums[i]
        return output
